amr-wind/wind_energy/actuator/disk/disk_spreading.H

```diff
@@ -0,0 +1,170 @@
+#ifndef DISK_SPREADING_H_
+#define DISK_SPREADING_H_
+
+#include "amr-wind/wind_energy/actuator/actuator_utils.H"
+#include "amr-wind/core/FieldRepo.H"
+#include <stdexcept>
+
+namespace amr_wind {
+namespace actuator {
+namespace ops {
+
+/**
+ * @brief  A collection of spreading functions
+ * This class allows for polymorphic spreading functions.
+ * The concept is based on Impossibly Fast C++ Delegates proposed here:
+ * https://www.codeproject.com/Articles/11015/The-Impossibly-Fast-C-Delegates
+ *
+ * @tparam T
+ */
+template <typename T>
+class SpreadingFunction
+{
+public:
+    void operator()(
+        T* actObj,
```

Comment:
Is it possible to pass a const reference instead of a pointer here? 

```C++
const T& actObj
```

Reply:
yeah it is. thanks good catch.